src/plugins/contributionReward/plugin.ts

```diff
@@ -0,0 +1,164 @@
+    contributionRewardParams {
+      id
+      votingMachine
+      voteParams {
+        id
+        queuedVoteRequiredPercentage
+        queuedVotePeriodLimit
+        boostedVotePeriodLimit
+        preBoostedVotePeriodLimit
+        thresholdConst
+        limitExponentValue
+        quietEndingPeriod
+        proposingRepReward
+        votersReputationLossRatio
+        minimumDaoBounty
+        daoBountyConst
+        activationTime
+        voteOnBehalf
+      }
+    }`
+  }
+
+  public static itemMap(arc: Arc, item: any): ContributionReward | null {
+    if (!item) {
+      return null
+    }
+
+    let name = item.name
+    if (!name) {
+
+      try {
+        name = arc.getContractInfo(item.address).name
+      } catch (err) {
+        if (err.message.match(/no contract/ig)) {
+          // continue
+        } else {
+          throw err
+        }
+      }
+    }
+
+    const contributionRewardParams = item.contributionRewardParams && {
+      voteParams: mapGenesisProtocolParams(item.contributionRewardParams.voteParams),
+      votingMachine: item.contributionRewardParams.votingMachine
+    }
+    
+    return new ContributionReward(arc, {
+        address: item.address,
+        canDelegateCall: item.canDelegateCall,
+        canManageGlobalConstraints: item.canManageGlobalConstraints,
+        canRegisterPlugins: item.canRegisterSchemes,
+        canUpgradeController: item.canUpgradeController,
+        dao: item.dao.id,
+        id: item.id,
+        name,
+        numberOfBoostedProposals: Number(item.numberOfBoostedProposals),
+        numberOfPreBoostedProposals: Number(item.numberOfPreBoostedProposals),
+        numberOfQueuedProposals: Number(item.numberOfQueuedProposals),
+        paramsHash: item.paramsHash,
+        pluginParams: contributionRewardParams,
+        version: item.version
+      }
+    )
+  }
+
+  public state(apolloQueryOptions: IApolloQueryOptions = {}): Observable<IContributionRewardSchemeState> {
+    const query = gql`query SchemeStateById
+      {
+        controllerScheme (id: "${this.id}") {
+          ...PluginFields
+        }
+      }
+      ${Plugin.baseFragment}
+    `
+    const itemMap = (item: any) => ContributionReward.itemMap(this.context, item)
+    return this.context.getObservableObject(query, itemMap, apolloQueryOptions) as Observable<IContributionRewardSchemeState>
+  }
+
+  public async createProposalTransaction(options: IProposalCreateOptionsCR): Promise<ITransaction> {
+    options.descriptionHash = await this.context.saveIPFSData(options)
+  
+    if (options.plugin === undefined) {
+      throw new Error(`Missing argument "plugin" for ContributionReward in Proposal.create()`)
+    }
+  
+    return {
+      contract: this.context.getContract(options.plugin),
+      method: 'proposeContributionReward',
+      args: [
+        options.dao,
+        options.descriptionHash || '',
+        options.reputationReward && options.reputationReward.toString() || 0,
```

Comment:
I think these evaluate to a boolean (like my last comment), would be worth a test in a typescript sandbox to see what the result is.

Reply:
I just checked this in repl.it and we are fine doing this